raft/multinode_test.go

```diff
@@ -476,3 +476,104 @@ func TestMultiNodeStatus(t *testing.T) {
 		t.Errorf("expected nil status, got %+v", status)
 	}
 }
+
+// TestMultiNodePerGroupID tests that MultiNode may have a different
+// node ID for each group, if and only if the Config.ID field is
+// filled in when calling CreateGroup.
+func TestMultiNodePerGroupID(t *testing.T) {
+	storage := NewMemoryStorage()
+	mn := StartMultiNode(1)
+
+	// Maps group ID to node ID.
+	groups := map[uint64]uint64{
+		1: 1,
+		2: 10,
+	}
+
+	// Group 1 uses the default node ID, 1.
+	err := mn.CreateGroup(1, newTestConfig(1, nil, 10, 1, storage),
```

Comment:
Actually, I'm going to change this so that we use `StartMultiNode(0)`, and it is an error to have a non-zero ID in both places. I was trying to allow for a mix of per-node and per-group IDs, but that's just silly. Anyone who wants per-group IDs is going to want them all the time.